pycbc/events/single.py

```diff
@@ -158,43 +198,62 @@ def check(self, trigs, data_reader):
         # Apply cuts to trigs before clustering
         # Cut on snr so that triggers which could not reach newsnr
         # threshold do not have newsnr calculated
+        if 'psd_var_val' in trigs:
+            # We should apply the PSD variation rescaling, as this can
+            # re-weight the SNR to be above SNR
+            trig_chisq = trigs['chisq'] / trigs['psd_var_val']
+            trig_snr = trigs['snr'] / (trigs['psd_var_val'] ** 0.5)
+        else:
+            trig_chisq = trigs['chisq']
+            trig_snr = trigs['snr']
+
         valid_idx = (trigs['template_duration'] >
                      self.thresholds['duration']) & \
-                    (trigs['chisq'] <
+                    (trig_chisq <
                      self.thresholds['reduced_chisq']) & \
-                    (trigs['snr'] >
+                    (trig_snr >
                      self.thresholds['newsnr'])
         if not np.any(valid_idx):
             return None
 
-        cutdurchi_trigs = {k: trigs[k][valid_idx] for k in trigs}
+        cut_trigs = {k: trigs[k][valid_idx] for k in trigs}
 
-        # This uses the pycbc live convention of chisq always meaning the
-        # reduced chisq.
-        nsnr_all = ranking.newsnr(cutdurchi_trigs['snr'],
-                                  cutdurchi_trigs['chisq'])
-        nsnr_idx = nsnr_all > self.thresholds['newsnr']
-        if not np.any(nsnr_idx):
+        # Convert back from the pycbc live convention of chisq always
+        # meaning the reduced chisq.
+        trigsc = copy.copy(cut_trigs)
+        trigsc['ifo'] = self.ifo
+        trigsc['chisq'] = cut_trigs['chisq'] * cut_trigs['chisq_dof']
+        trigsc['chisq_dof'] = (cut_trigs['chisq_dof'] + 2) / 2
```

Comment:
This is the same as [this part of the coinc version](https://github.com/gwastro/pycbc/blob/v2.1.3/pycbc/events/coinc.py#L1086)